### PR TITLE
Support Sky UX Icons in Dropdown

### PR DESCRIPTION
**Pull request by johnhwhite**
Allow SkyUX Icons in dropdown

[Bug 1754712](https://blackbaud.visualstudio.com/Products/_workitems/edit/1754712)

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-support-icons](https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-support-icons)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-support-icons](https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-support-icons)

File: src/app/public/modules/dropdown/dropdown.component.spec.ts [hunk 1]
```diff
@@ -146,6 +173,20 @@ describe('Dropdown component', function () {
     expect(button).toHaveCssClass('sky-btn-default');
   }));
 
+  it('should change theme', fakeAsync(() => {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/dropdown/dropdown.component.spec.ts:176]: What's this actually testing? We already know the theme service works... is this necessary?

File: src/app/public/modules/dropdown/dropdown.component.ts [hunk 2]
```diff
@@ -253,6 +258,14 @@ export class SkyDropdownComponent implements OnInit, OnDestroy {
       .subscribe((message: SkyDropdownMessage) => {
         this.handleIncomingMessages(message);
       });
+
+    this.themeSvc.settingsChange
+      .pipe(
+        takeUntil(this.ngUnsubscribe)
+      )
+      .subscribe(() => {
+        this.changeDetector.markForCheck();
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/dropdown/dropdown.component.ts:267]: I'm guessing this is necessary to get the themeIf's to properly trigger? Would you mind adding a comment here to explain?

File: src/app/visual/dropdown/dropdown-visual.component.html [hunk 3]
```diff
@@ -73,6 +73,7 @@ <h2>Dropdown next to position:relative content</h2>
   <div
     style="position: relative; width: 300px; height: 100px; background-color: red"
   >
+  </div>
```
Comment (Blackbaud-AlexKingman) [inline src/app/visual/dropdown/dropdown-visual.component.html:76]: Nice catch! Weird that this wasn't caught by our build...

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Looks good. Are we doing the work to make `buttonType` support skyux icons in a separate PR?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-support-icons](https://developer.blackbaud.com/skyux-popovers?_sp.spa=pr-support-icons)

File: src/app/public/modules/dropdown/dropdown.component.spec.ts (hunk 1 above)
Comment (johnhwhite) [reply src/app/public/modules/dropdown/dropdown.component.spec.ts:180]: Updated to actually test something.

File: src/app/public/modules/dropdown/dropdown.component.ts (hunk 2 above)
Comment (johnhwhite) [reply src/app/public/modules/dropdown/dropdown.component.ts:268]: Added a comment. With OnPush change detection, this is necessary for conditional `<sky-icon>` elements to render when the theme changes.

**Review by Blackbaud-AlexKingman — APPROVED** [review]

**Pull request merged**